core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/GoogleProjectRegistrationApiService.scala

```diff
@@ -0,0 +1,38 @@
+package org.broadinstitute.dsde.rawls.webservice
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server
+import akka.http.scaladsl.server.Directives._
+import io.opentelemetry.context.Context
+import org.broadinstitute.dsde.rawls.googleProject.GoogleProjectRegistrationService
+import org.broadinstitute.dsde.rawls.model.GoogleProjectRegistrationJsonSupport$._
+import org.broadinstitute.dsde.rawls.model.{GoogleProjectRegistration, RawlsRequestContext}
+import org.broadinstitute.dsde.rawls.openam.UserInfoDirectives
+
+import scala.concurrent.ExecutionContext
+
+trait GoogleProjectRegistrationApiService extends UserInfoDirectives {
+  implicit val executionContext: ExecutionContext
+
+  val googleProjectRegServiceConstructor: RawlsRequestContext => GoogleProjectRegistrationService
+
+  def googleProjectRegistrationRoutes(otelContext: Context = Context.root()): server.Route =
+    requireUserInfo(Option(otelContext)) { userInfo =>
+      val ctx = RawlsRequestContext(userInfo, Option(otelContext))
+      path("googleProjects") {
+        put {
+          entity(as[GoogleProjectRegistration]) { entity =>
+            complete {
+              googleProjectRegServiceConstructor(ctx)
+                .registerGoogleProject(
+                  entity
+                )
+                .map(w => StatusCodes.Created -> w)
```

Comment:
should this return 200 in cases where the project was already registered, or is it always acceptable to return 201?

Reply:
Ah it should be a [200](https://github.com/broadinstitute/rawls/pull/3217#discussion_r1996051524), thanks for noticing